### PR TITLE
Do not call vararg C functions from Fortran (JIRA CGNS-179)

### PR DESCRIPTION
**Pull request by pini-gh**
Please see rationals on JIRA CGNS-179:
https://cgnsorg.atlassian.net/browse/CGNS-179

### CODE REVIEW

File: src/cg_ftoc.c [hunk 1]
```diff
@@ -2949,19 +2921,9 @@ CGNSDLL void FMNAME(cg_gorel_f, CG_GOREL_F)(cgint_f *fn, cgint_f *ier, ...)
 
      /* read arguments */
     for (n = 0; n < CG_MAX_GOTO_DEPTH; n++)  {
-#ifdef _CRAY
```
Comment (MicK7) [inline src/cg_ftoc.c]: Why remove CRAY and WINDOWS parts

**Comment by MicK7** [comment]
Thanks for the proposal. gfortran test are failing on travis.

File: src/cg_ftoc.c (hunk 1 above)
Comment (pini-gh) [reply src/cg_ftoc.c]: As I understand it these parts are not needed anymore when *not* calling C varargs from Fortran code.

File: src/cgns_f.F90 [hunk 2]
```diff
@@ -5046,6 +5055,332 @@ END SUBROUTINE cgio_set_label_f
...
+     INTEGER, OPTIONAL :: index18
+     CHARACTER(*), OPTIONAL :: name19
+     INTEGER, OPTIONAL :: index19
+     CHARACTER(*), OPTIONAL :: name20
+
+     IF (.NOT. PRESENT(index1)) THEN
+        CALL cg_goto_f1(fn, B, ier, name1, 0)
+        RETURN
+     ELSE
+        PRINT '(I3)', index1
```
Comment (MicK7) [inline src/cgns_f.F90]:
```suggestion

```

**Comment by MicK7** [comment]
LGTM

**Comment by pini-gh** [comment]
> Keep in mind that we test the following fortran compilers: NAG, PGI, GNU, Intel, xl, cray, sun. On windows we only test Intel (manually). I would need to make sure all tests pass before merging this. So is this with gfortran 10, isn't it still in developement?

The change that breaks calling vararg C functions from Fortran code on powerpc was introduced into the GCC 9 development branch. Please see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=92361.
> 
> Can you do your pull request to the CGNS-179 branch so that I can run it through our daily tests. I'll try it with cray and windows.

Doing that.

**Comment by MicK7** [comment]
@pini-gh @brtnfld  I am moving the commits to CGNS-179 branch. No need for a new PR.

**Comment by MicK7** [comment]
The changes are now in CGNS-179 branch.

**Comment by brtnfld** [comment]
Thanks, I'll switch to testing that branch.

**Comment by MicK7** [comment]
Merge Request transfered to #188

**Pull request closed without merging**